docs/pages/machine-id/architecture.mdx

```diff
@@ -0,0 +1,148 @@
+  Machine ID agent to initially authenticate as the bot user. If an existing
+  token is not specified, then a static single-use token will be created by the
+  Auth Server.
+
+## `tbot`
+
+`tbot` is the binary that acts as the agent for Machine ID.
+
+On initial load, `tbot` uses the [configured join method](#onboarding) to obtain
+a set of credentials for the bot user from the Teleport Auth Service. It can 
+then use these credentials to communicate with the Teleport Auth Service as the 
+bot.
+
+On a configured regular period `tbot` then:
+
+1. Renews these bot user credentials
+2. For each destination provided in `tbot`'s configuration:
+   1. Obtains [role impersonated](#role-impersonation) credentials for the roles
+      specified in the destination configuration from the Auth Service using the
+      bot user credentials.
+   2. Outputs these role impersonated credentials in various formats.
+
+Concurrently to this, `tbot` monitors the Teleport certificate authorities to
+detect certificate rotations. When this occurs, it triggers additional renewals
+to ensure that `tbot` has certificates signed by the newly rotated certificate
+authority.
+
+## Role Impersonation
+
+At the core of Machine ID is the concept of role impersonation.
+
+Role Impersonation allows a user to generate credentials with a set of requested
+roles. The user does not have to hold these roles, but must have been granted 
+permission to impersonate them. The impersonated credentials still include
+the username of the user that generated them, so actions can be attributed to 
+the user.
+
+These credentials can then be used to complete any action that is allowed by the
+role's configured permissions. 
+
+In the case of Machine ID, the bot user is assigned a bot role, which includes 
+permissions to impersonate the roles that the user has configured.
+
+## Onboarding
+
+Onboarding is the process by which `tbot` initially authenticates with the 
+Teleport auth service in order to receive credentials.
+
+Machine ID leverages the existing Token resource within Teleport, with the
+token containing an additional `botName` field that identifies the bot user
+associated with the token. 
+
+Machine ID currently supports two methods of joining that have some key
+differences.
+
+Static Token:
+
+- The name of the token is the only value needed to join. This value is a
+  secret and must be handled securely.
+- Once used, the token resource self-destructs. This means it can only be used 
+  to join a single bot to a Teleport cluster.
+- The certificates exchanged for the token are 
+  [renewable](#renewable-certificates).
+
+
+Dynamic join tokens (e.g AWS IAM):
+
+- These tokens rely on some kind of external system that allows the bot to
+  prove it is allowed to join the cluster. The name of the token identifies the
+  Token resource in Teleport that contains the configuration.
+- The token can be used to join as many bots as you want.
```

Comment:
Maybe it's clear as is, but would it help to specifically call out these don't have the self-destruct behavior?

Reply:
I'm a little concerned that exhaustively listing these will eventually be verbose (as we will support IAM, GCP, Github, Gitlab and more in coming months...). We should have another page on the docs that lists these exhaustively and documents their configuration that we can link out to.

Reply:
f4ba44362